docs/docs/policies.mdx

```diff
@@ -689,36 +993,59 @@ default restart, label `2` a greeting and `3` indicates goodbye.
-when predicting an action. The `TrackerFeaturizer` iterates over tracker
-states and calls a `SingleStateFeaturizer` for each state to create numeric
-input features for a policy.
-The target labels correspond to bot actions or bot utterances
-represented as index in a list of all possible actions.
+A policy can be trained to learn two kinds of labels -
+
+1. The next most appropriate action to be triggered by the assistant. For example, [`TEDPolicy`](#ted-policy) is trained to do this.
+2. The next most likely intent that a user can express. For example, [`IntentTEDPolicy`](#intenttedpolicy) is trained to do this.
+
+Hence, a tracker can be featurized to learn one of the above kind of labels.
+Depending on the policy, the target labels correspond to bot actions or bot utterances
+represented as an index in a list of all possible actions or
+set of intents represented as an index in a list of all possible intents.
 
-There are two different tracker featurizers:
+There are three different tracker featurizers:
 
 ##### 1. Full Dialogue
 
 `FullDialogueTrackerFeaturizer` creates a numerical representation of
 stories to feed to a recurrent neural network where the whole dialogue
 is fed to a network and the gradient is backpropagated from all time steps.
-The smaller dialogues are padded with `0` for all features.
+The target label is the most appropriate bot action or bot utterance which should be triggered in the
+context of the conversation.
+The `TrackerFeaturizer` iterates over tracker
+states and calls a `SingleStateFeaturizer` for each state to create numeric input features for a policy.
+
 
 ##### 2. Max History
 
-`MaxHistoryTrackerFeaturizer` creates an array of previous tracker
-states for each bot action or bot utterance, with the parameter
+`MaxHistoryTrackerFeaturizer` operates very similarly to `FullDialogueTrackerFeaturizer` as
+it creates an array of previous tracker states for each bot action or bot utterance but with the parameter
 `max_history` defining how many states go into each row of input features.
 If `max_history` is not specified, the algorithm takes
 the whole length of a dialogue into account.
-The smaller dialogues are padded with `0` for all features.
 Deduplication is performed to filter out duplicated turns (bot actions
 or bot utterances) in terms of their previous states.
 
 For some algorithms a flat feature vector is needed, so input features
 should be reshaped to `(num_unique_turns, max_history * num_input_features)`.
 
+##### 3. Intent Max History
+
+`IntentMaxHistoryTrackerFeaturizer` inherits from `MaxHistoryTrackerFeaturizer`. Since, it is used by
+[`IntentTEDPolicy`](#intenttedpolicy), the target labels that it creates is the set of intents that can be
+expressed by a user in the context of a conversation tracker. Unlike
+other tracker featurizers, there can be multiple target labels. Hence, it pads the
+list of target labels with `-1` values on the right to return an equally sized list of target labels
```

Comment:
Shouldn't we use the constant (or just write "a constant") here, instead of `-1`?

Reply:
Changed it to ```list of target labels with a constant value (`-1`) on the right```